app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,23 @@
+class CommentsController < ApplicationController
+  def new
+    @companies = Comapny.all
+    @job       = Job.all
+    @comment   = Comment.new
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.job_id = params[:job_id]
+    if @comment.save
+      flash[:success] = "Comment created!"
+      redirect_to company_job_path(@comment.job.company, @comment.job)
+    else
+      redirect_to company_job_path(@comment.job.company, @comment.job)
```

Comment:
same - handle your errors so a user sees a nice error message rather than being redirected to the same place regardless if the data the user enters saves or not